src/python/pants/reporting/zipkin_reporter.py

```diff
@@ -19,19 +21,43 @@
 
 
 class HTTPTransportHandler(BaseTransportHandler):
-  def __init__(self, endpoint):
+  def __init__(self, endpoint, zipkin_spans_dir, encoding):
     self.endpoint = endpoint
+    self.zipkin_spans_dir = zipkin_spans_dir
+    self.file_count = 0
+    self.encoding = encoding
 
   def get_max_payload_bytes(self):
     return None
 
   def send(self, payload):
     try:
-      requests.post(
-        self.endpoint,
-        data=payload,
-        headers={'Content-Type': 'application/x-thrift'},
-      )
+      if os.path.exists(self.zipkin_spans_dir):
```

Comment:
Let's remove this check - we generally trust that people don't mess with the workdir underneath us.

If we want to keep it, I'd probably phrase it more as:
```python
if not os.path.exists(self.zipkin_spans_dir):
  log.error("Not uploading zipkin spans because directory {} got deleted".format(self.zipkin_spans_dir))
  return
```

Two things about that:
1. Doing the "if not - early return" thing means you don't end up indenting all of the code a bunch (which means you have less context to think about when reading it)
2. Logging that we're not doing something because of something we didn't expect, which indicates something went wrong somewhere